aptos-move/framework/src/extended_checks.rs

```diff
@@ -479,8 +489,59 @@ impl<'a> ExtendedChecker<'a> {
                 .fun_attributes
                 .entry(fun.get_simple_name_string().to_string())
                 .or_default()
-                .push(KnownAttribute::randomness());
+                .push(KnownAttribute::randomness(randomness_annotation.max_gas));
+        }
+    }
+
+    fn get_randomness_max_gas_declaration(
+        &self,
+        fun: &FunctionEnv,
+    ) -> Result<Option<RandomnessAnnotation>, String> {
+        for attr in fun.get_attributes() {
+            let Attribute::Apply(_, name, sub_attrs) = attr else {
+                continue;
+            };
+            if self.env.symbol_pool().string(*name).as_str() != RANDOMNESS_ATTRIBUTE {
+                continue;
+            }
+            if sub_attrs.len() >= 2 {
```

Comment:
Should this be !=1, or at least >= 1?

Reply:
we only support `#[randomness]` and `#[randomness(max_gas=x)]`, so only 0 and 1 are valid?

Reply:
Thanks!